README.md

```diff
@@ -1 +1,404 @@
+
+* With key: [http://api.giphy.com/v1/gifs/search?q=funny+cat&api_key=dc6zaTOxFJmzC](http://api.giphy.com/v1/gifs/search?q=funny+cat&api_key=dc6zaTOxFJmzC)
+
+**It is very important that you not push your API keys to a public Github repo.**
+
+## AJAX (5 mins)
+
+**So we know what an API is, now what?**
+
+How can we use an API to dynamically manipulate the DOM with the given data? In Rails, we would have to make a request through the server, and refresh the page with returned data.  Now we can do this on the client side! As you'll come to learn, this allows us to build single page applications that do not require refreshes.
+
+How do we do this?
+*AJAX* (Asynchronous Javascript and XML) is the method through which we are able to make the usual HTTP requests: 'GET' 'POST' 'PUT' 'PATCH' 'DELETE' (depending on the access we have) to a given API.
+
+## Break (10 mins)
+
+## GET data from an external API using AJAX
+
+### I Do: GET example (10 mins)
+
+Let's build a very simple app that posts a movie title and poster after searching for it!
+[OMDB movie search](https://github.com/joe-gz/omdb-api)
+
+The starter code is linked above. It contains a basic HTML/CSS/JS setup. If you open up the HTML in the browser, you will notice that searching for something returns no results.
+
+Let's go ahead and add in the AJAX request:
+
+```js
+// get value from search input field
+var keyword = $("input[name='keyword']").val();
+var url = "https://www.omdbapi.com/?t="+keyword
+$.ajax({
+  url: url,
+  type: "GET",
+  dataType: "json"
+}).done ( function(response){
+  console.log(response);
+  // call movie function below to append movie titles
+  movie(response);
+}).fail ( function (){
+  console.log("fail");
+}).always( function(){
+  console.log("Something happens");
+})
+```
+
+**What does all of this mean?**
+
+$.ajax takes an object as an argument with at least three key-value pairs...
+(1) The URL endpoint for the JSON object.
+(2) Type of HTTP request.
+(3) Datatype. Usually JSON.
+
+The `url` that we've included is the api url from OMDB, with the added keyword that we are searching for. The `type` is simply the verb we want performed; in this case GET. And, the `datatype` will typically be `json`.
+
+### Promises
+
+Following `datatype`, there are three methods that we call promises.  In short, these tell the `$.ajax` request what to do if the request is successful, fails, or both.
+
+* `.done` requires a callback that determines what we do after a successful AJAX call
+* `.fail` requires a callback that determines what we do after an unsuccessful AJAX call
+* `.always` requires a callback that determines what we do regardless of a successful or unsuccessful call
+
+**Bonus**: you'll notice that if you simply search for "Star Wars", it doesn't really work! why is that? Is there another url you can use from OMDB API that would return all Star Wars movies?
```

Comment:
Could this bonus be moved to an exercise, so that when students are working on their own and they finish early they have time to try this out?